scrape/scrape.go

```diff
@@ -877,6 +883,9 @@ type scrapeCache struct {
 	metadata map[string]*metaEntry
 
 	metrics *scrapeMetrics
+
+	nhcbLabels  map[uint64]labels.Labels
```

Comment:
If we're going to reset these after each `append` call, could they be just a local variable in `append`?